cli_tools/common/vdisk/qemu_img.go

```diff
@@ -0,0 +1,107 @@
+//  Copyright 2020 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package vdisk
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/files"
+)
+
+// Format is an enum type for representing VM disk encoding formats.
+type Format string
+
+const (
+	// FormatUnknown means that qemu-img could not determine the file's format.
+	FormatUnknown Format = "unknown"
+	// FormatVMDK means that the file uses the vmdk file format.
+	FormatVMDK Format = "vmdk"
+	// FormatVHDX means that the file uses the vhdx file format.
+	FormatVHDX Format = "vhdx"
+	// FormatVPC means that the file uses the vpc file format.
+	FormatVPC Format = "vpc"
+	// FormatVDI means that the file uses the vdi file format.
+	FormatVDI Format = "vdi"
+	// FormatQCOW2 means that the file uses the qcow2 file format.
+	FormatQCOW2 Format = "qcow2"
+	// FormatRAW means that the file uses the raw file format.
+	FormatRAW Format = "raw"
```

Comment:
The list of formats qemu-img supports is longer than this. Why is the list limited to these?

Reply:
Good question. I used three strategies for generating the list:
1. The formats that we document supporting. These were the places that I found such lists; interestingly, I didn't 
  * https://cloud.google.com/sdk/gcloud/reference/compute/images/export
  * https://cloud.google.com/sdk/gcloud/reference/compute/images/import
2. I looked at a couple weeks of import history.
3. I ran `qemu-img --help`, and these were the formats that I recognized as industry virtual disk formats

Any formats that we document supporting that I missed?

Reply:
For import, we never documented formats since we use qemu-img. We mostly just shown examples ("like vmdk, vhd...").

Reply:
This isn't limiting what's possible, but rather the set of values that are logged (to avoid table fields that are classified as free-form strings). Do you have a recommendation to move forward?

Reply:
This is the full list of formats from qemu-img --help:

blkdebug blklogwrites blkreplay blkverify bochs cloop copy-on-read dmg file ftp ftps gluster host_cdrom host_device http https iscsi iser luks nbd nfs null-aio null-co nvme parallels qcow qcow2 qed quorum raw rbd replication sheepdog ssh throttle vdi vhdx vmdk vpc vvfat

I have no idea what half of these mean, but maybe we can just validate the input value against this list (no need for constants) and if it's not in there, return Unknown?

Reply:
Updated to validate against the list of formats returned by `qemu-img --help`